plugins/buildingplan.cpp

```diff
@@ -959,6 +969,8 @@ DFhackCExport command_result plugin_onstatechange(color_ostream &out, state_chan
 {
     switch (event) {
     case SC_MAP_LOADED:
+        buildingplan_place_hook::invalidateStatics();
+        buildingplan_query_hook::invalidateStatics();
         planner.reset();
```

Comment:
Ah, so the `planner.reset()` call was invalidating the iterators - fixing it here makes sense to me, then.